g3w-suite/g3w-admin
# :sparkles: Add support to qgis `'postgresraster'` layer

Closes: #651

## How to test

0. get latest [PostGIS](https://postgis.net/)

1. create a database and add postgis_raster extension:

```sql
create extension postgis_raster;
```

![image](https://github.com/g3w-suite/g3w-admin/assets/9614886/e1391fb0-c500-4b70-9cfa-9a58f9b65678)

2. create a schema `g3w_suite` and run inside the database the following sql: [bluemarble.zip](https://github.com/g3w-suite/g3w-admin/files/13332260/bluemarble.zip)

![image](https://github.com/g3w-suite/g3w-admin/assets/9614886/6584eae5-a54a-4af9-bacc-160110fa629a)

3. open qgis and check that sql has created a simple `bluemarble` raster tif layer with **EPSG:4326** into db:

![Screenshot_20231113_092106](https://github.com/g3w-suite/g3w-admin/assets/196809/ce0bb56d-c641-453f-9ff9-5813002932eb)

## Additional notes

If you are working in a mixed setup (ie. a local [postgis](https://postgis.net/) instance + a [g3w-suite-docker](https://github.com/g3w-suite/g3w-suite-docker) container), you should add an `extra_hosts` directive within your `docker-compose-dev.yml` to make your local postgres databases accessible from both sides:

![image](https://github.com/g3w-suite/g3w-admin/assets/9614886/ade856d2-99ec-4024-ab0d-7c631cfa67e8)


```yaml

  g3w-suite:
    image: g3wsuite/g3w-suite:dev

    ...

    extra_hosts:
      - "postgis16:host-gateway"
```

taking care to edit your `hosts` file accordingly:

```sh
# Added for G3W-SUITE docker
127.0.0.1 postgis16
```

**For more info:**

- https://docs.docker.com/compose/compose-file/compose-file-v3/#extra_hosts
